File: backendNodeJS/src/routes/todo.routes.js
import { Router } from "express";
import { authValidation } from "../middlewares/auth.middleware.js";
import {
    getToDoCategory,
    createToDoCategory,
    getToDo,
    createToDo,
    toggleToDo,
    deleteToDoCategory,
    deleteToDo,
} from "../controllers/todo.controller.js";

export const todoRouter = Router();

todoRouter.route("/getToDoCategory").get(authValidation, getToDoCategory);
todoRouter
    .route("/createToDoCategory")
    .post(authValidation, createToDoCategory);
todoRouter
    .route("/deleteToDoCategory")
    .delete(authValidation, deleteToDoCategory);
todoRouter.route("/getToDo").get(authValidation, getToDo);
todoRouter.route("/createToDo").post(authValidation, createToDo);
todoRouter.route("/deleteToDo").delete(authValidation, deleteToDo);
todoRouter.route("/toggleToDo").get(authValidation, toggleToDo);

File: backendFlask/src/routes/todo_routes.py
from flask import Blueprint
from controllers.todo_controller import (
    get_todo_category,
    create_todo_category,
    get_todo,
    create_todo,
    toggle_todo,
    delete_todo_category,
    delete_todo,
)
from middlewares.auth_middleware import auth_validation

todo_bp = Blueprint("todo", __name__)

todo_bp.route("/getToDoCategory", methods=["GET"])(auth_validation(get_todo_category))
todo_bp.route("/createToDoCategory", methods=["POST"])(
    auth_validation(create_todo_category)
)
todo_bp.route("/deleteToDoCategory", methods=["DELETE"])(
    auth_validation(delete_todo_category)
)
todo_bp.route("/getToDo", methods=["GET"])(auth_validation(get_todo))
todo_bp.route("/createToDo", methods=["POST"])(auth_validation(create_todo))
todo_bp.route("/deleteToDo", methods=["DELETE"])(auth_validation(delete_todo))
todo_bp.route("/toggleToDo", methods=["GET"])(auth_validation(toggle_todo))
